client/my-sites/checkout/checkout/stripe-elements-payment-box.jsx

```diff
@@ -0,0 +1,260 @@
+			return;
+		}
+		debug( 'loading stripe.js...' );
+		loadScript( url, function( error ) {
+			if ( error ) {
+				debug( 'stripe.js script ' + error.src + ' failed to load.' );
+				return;
+			}
+			debug( 'stripe.js loaded!' );
+			setStripeJs( window.Stripe( apiKey ) );
+		} );
+	}, [ url, apiKey ] );
+	return stripeJs;
+}
+
+async function submitPaymentForm( { stripe, paymentDetails, cart, transaction } ) {
+	const stripePaymentMethod = await createStripePaymentMethod( stripe, paymentDetails );
+	return sendStripePaymentMethodToStore( {
+		cart,
+		transaction,
+		paymentDetails,
+		stripePaymentMethod,
+	} );
+}
+
+async function createStripePaymentMethod( stripe, paymentDetails ) {
+	debug( 'creating payment method...', paymentDetails );
+	const { paymentMethod, error } = await stripe.createPaymentMethod( 'card', {
+		billing_details: paymentDetails,
+	} );
+	debug( 'payment method creation complete', paymentMethod, error );
+	if ( error ) {
+		// Note that this is a promise rejection
+		throw new Error( error.message );
+	}
+	return paymentMethod;
+}
+
+async function handleStripeAction( stripe, stripeResponse, orderId, redirectTo ) {
+	const {
+		error: errorAction
+	} = await stripe.handleCardPayment(
+		stripeResponse.payment_intent_client_secret
+	);
+
+	if ( errorAction ) {
+		// Show error from Stripe.js in payment form
+	} else {
+		const origin = getLocationOrigin( window.location );
+		const redirectPath = redirectTo().replace( ':receiptId', '');
+		page( origin + redirectPath + 'pending/' + orderId );
+	}
+}
+
+async function sendStripePaymentMethodToStore( {
+	cart,
+	transaction,
+	paymentDetails,
+	stripePaymentMethod,
+} ) {
+	const dataForApi = {
+		payment: Object.assign( {}, paymentDetails, {
+			paymentMethod: paymentMethodClassName( 'stripe' ),
+			paymentKey: stripePaymentMethod.id,
+			successUrl: '/',
+			cancelUrl: '/',
+		} ),
+		cart,
+		domain_details: transaction.domainDetails,
+	};
+
+	return wpcom.undocumented().transactions( 'POST', dataForApi );
+}
+
+function StripeElementsForm( {
+	translate,
+	stripe,
+	cart,
+	transaction,
+	children,
+	countriesList,
+	redirectTo,
+	presaleChatAvailable,
+} ) {
+	// TODO: allow disabling the payment button during processing or when not ready
+	const [ cardholderName, setCardholderName ] = useState( '' );
+	const onNameChange = event => setCardholderName( event.target.value );
+	const handleSubmit = event => {
+		event.preventDefault();
+		submitPaymentForm( {
+			stripe,
+			cart,
+			transaction,
+			paymentDetails: {
+				name: cardholderName,
+			},
+		} )
+			.then( result => {
+				if ( result.message )  {
+					handleStripeAction( stripe, result.message, result.order_id, redirectTo );
+				} else if ( result.redirect_url ) {
+					notices.success(
+						translate( 'Redirecting you to the payment partner to complete the payment.' )
+					);
+					window.location.href = result.redirect_url;
+				}
+				// TODO: what if there is no redirect_url?
```

Comment:
We won't be using the redirect_Url  in this payment box since we won't be redirecting to an external URL. Stripe.js will use the `message` property we'll pass to `handleStripeAction` to take care of the authentication.

However, if there's no `message` property (and we get a success:true, and a receipt id), then we have a successful CC payment that doesn't require authentication, and we just need to redirect to the success page.

Reply:
At this point, I'm not sure if it's maybe worth to reuse the transactionSteps/submitTransaction logic from the regular cc form